Add error message under input field.

PARTHVATALIYA opened the pull request:
Trac ticket: https://core.trac.wordpress.org/ticket/60969

Add error message to URL input field under the add custom link section.
Add CSS to the error message.

PARTHVATALIYA committed:
Add error message under input field.

PARTHVATALIYA committed:
Remove debug code.

joedolson commented on src/js/_enqueues/lib/nav-menu.js:
```diff
@@ -1103,12 +1103,38 @@
 
 			$('#add-custom-links input[type="text"]').on( 'keypress', function(e){
 				$('#customlinkdiv').removeClass('form-invalid');
+				$('#custom-menu-item-url').hide();
```
This should be `#custom-url-error`.

joedolson commented on src/js/_enqueues/lib/nav-menu.js:
```diff
@@ -1103,12 +1103,38 @@
+						.attr('aria-describedby', 'custom-url-error');
+
+					var errorText = wp.i18n.__('No content entered. Please enter a valid link.');
+					errorMessage.show();
```
I know I suggested this text, but given the flexibility in what the possible values can be, probably need to make this more general yet. Perhaps "Content is required for the link's `href` attribute."

joedolson reviewed (dismissed):
There are also some misc. code style issues in the JS, mostly surrounding use of spaces - in general the content inside parentheses should have spaces, e.g. `$( '#submit-customlinkdiv' )`. 

But this is generally good; just needs some minor polish.

PARTHVATALIYA committed:
Update error message for screen reader.

PARTHVATALIYA commented:
@joedolson I made the changes you suggested can you please review it.

Thank You!

joedolson committed:
Merge branch 'trunk' into pr/8273

joedolson committed:
Various improvements

1) Spacing; 2) Remove aria-invalid and aria-describedby when other invalid classes and errors removed. 3) Get error message from content rather than duplicating the message.

joedolson committed:
Support error messages in customizer nav menus.

joedolson committed:
Fix quotes

joedolson committed:
Match the URL validation from customizer

joedolson committed:
Copy validation into pre-add test; change error message.

joedolson committed:
Declare the urlRegex variable.

joedolson committed:
String no longer requires double quoting.

joedolson committed:
Omitted comma.

joedolson commented:
Closed with https://core.trac.wordpress.org/changeset/59948
